library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/MainActivity.kt

```diff
@@ -136,12 +136,11 @@ internal class MainActivity :
     }
 
     private fun exportTransactions(fileName: String, block: suspend (List<HttpTransaction>) -> Sharable) {
+        val applicationContext = this.applicationContext
         lifecycleScope.launch {
             val transactions = viewModel.getAllTransactions()
             if (transactions.isNullOrEmpty()) {
-                Toast
-                    .makeText(this@MainActivity, R.string.chucker_export_empty_text, Toast.LENGTH_SHORT)
-                    .show()
+                showToast(applicationContext.getString(R.string.chucker_export_empty_text))
                 return@launch
             }
```

Comment:
I have a feeling it was this which caused the leak since it references the activity after launching a coroutine.

The other updates aren't required, and activity can be fine as you mentioned.  I like to always try and use application Context just as a best practice to avoid these sorts of leaks.

Reply:
Same feeling that this is the place.